assets/js/util.js

```diff
@@ -308,13 +321,61 @@ function callApy(options, endpoint) {
     var requestUrl = window.location.protocol + window.location.hostname +
         window.location.pathname + '?' + $.param(requestOptions.data);
 
+    apyRequestStartTime = Date.now();
+    apyRequestTimeout = setTimeout(function () {
+        displayInstallationNotification();
+        clearTimeout(apyRequestTimeout);
+    }, INSTALLATION_NOTIFICATION_INDIVIDUAL_DURATION_THRESHOLD);
+
     if(requestUrl.length > APY_REQUEST_URL_THRESHOLD_LENGTH) {
         requestOptions.type = 'POST';
         return $.ajax(requestOptions);
     }
     return $.jsonp(requestOptions);
 }
 
+function recordAPyRequestCompletion(requestDuration) {
+    cumulativeAPyRequestDuration += requestDuration;
+    if(lastNAPyRequestDurations.length === INSTALLATION_NOTIFICATION_REQUESTS_BUFFER_LENGTH) {
+        cumulativeAPyRequestDuration -= lastNAPyRequestDurations[0];
+        lastNAPyRequestDurations.shift();
+        lastNAPyRequestDurations.push(requestDuration);
+    }
+    else {
+        lastNAPyRequestDurations.push(requestDuration);
+    }
+
+    var averageRequestDuration = cumulativeAPyRequestDuration / lastNAPyRequestDurations.length;
+
+    if(requestDuration > INSTALLATION_NOTIFICATION_INDIVIDUAL_DURATION_THRESHOLD ||
+        averageRequestDuration > INSTALLATION_NOTIFICATION_CUMULATIVE_DURATION_THRESHOLD) {
+        displayInstallationNotification();
+    }
+}
+
+function displayInstallationNotification() {
+    if(installationNotificationShown) {
+        return;
+    }
+    installationNotificationShown = true;
+
+    $('#installationNotice').fadeIn('slow').removeClass('hide')
+        .delay(INSTALLATION_NOTIFICATION_DURATION)
+        .fadeOut('slow', hideInstallationNotification);
+
+    $('#installationNotice').mouseover(function () {
+        $(this).stop(true);
+    }).mouseout(function () {
+        $(this).animate()
+            .delay(INSTALLATION_NOTIFICATION_DURATION)
+            .fadeOut('slow', hideInstallationNotification);
+    });
+}
+
+function hideInstallationNotification() {
```

Comment:
Move this function into `displayInstallationNotification` since it isn't used anywhere else?